machineconfiguration/v1/types_machineosbuild.go

```diff
@@ -0,0 +1,208 @@
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+type MachineOSBuildList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	// items contains a collection of MachineOSBuild resources.
+	// +optional
+	Items []MachineOSBuild `json:"items"`
+}
+
+// MachineOSBuildSpec describes information about a build process primarily populated from a MachineOSConfig object.
+type MachineOSBuildSpec struct {
+	// machineConfig points to the rendered MachineConfig resource to be included in this image build.
+	// +required
+	MachineConfig MachineConfigReference `json:"machineConfig"`
+	// machineOSConfig references the MachineOSConfig resource that this image build extends.
+	// +required
+	MachineOSConfig MachineOSConfigReference `json:"machineOSConfig"`
+	// renderedImagePushSpec is set by the Machine Config Operator from the MachineOSConfig object this build is attached to.
+	// This field describes the location of the final image, which will be pushed by the build once complete.
+	// The format of the image push spec is: host[:port][/namespace]/name:<tag> or svc_name.namespace.svc[:port]/repository/name:<tag>.
+	// The length of the push spec must be between 1 to 447 characters.
+	// +required
+	RenderedImagePushSpec ImageTagFormat `json:"renderedImagePushSpec"`
+}
+
+// MachineOSBuildStatus describes the state of a build and other helpful information.
+// +kubebuilder:validation:XValidation:rule="has(self.buildEnd) ? has(self.buildStart) && timestamp(self.buildStart) < timestamp(self.buildEnd) : true",message="buildEnd must be after buildStart"
+type MachineOSBuildStatus struct {
+	// conditions are state related conditions for the build. Valid types are:
+	// Prepared, Building, Failed, Interrupted, and Succeeded.
+	// Once a Build is marked as Failed or Interrupted, no future conditions can be set.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	// +kubebuilder:validation:MaxItems=8
+	// +kubebuilder:validation:XValidation:rule="oldSelf.exists(x, x.type=='Failed') ? self==oldSelf : true",message="once a Failed condition is set, conditions are immutable"
+	// +kubebuilder:validation:XValidation:rule="oldSelf.exists(x, x.type=='Interrupted') ? self==oldSelf : true",message="once an Interrupted condition is set, conditions are immutable"
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+	// builder describes the image builder backend used for this build.
+	// +optional
+	Builder *MachineOSBuilderReference `json:"builder"`
+	// relatedObjects is a list of references to ephemeral objects such as ConfigMaps or Secrets that are meant to be consumed while the build process runs.
+	// After a successful build or when this MachineOSBuild is deleted, these ephemeral objects should be deleted.
```

Comment:
Should is wooly, be direct, they will be removed, if they aren't, it's a bug
```suggestion
	// After a successful build or when this MachineOSBuild is deleted, these ephemeral objects will be removed.
```